src/auth/permissions/index.ts

```diff
@@ -0,0 +1,173 @@
+import * as knex from 'knex';
+import { PermissionsInitialiser, GetPermissionIds, GetRoleId } from './permissions';
+
+const getPermissionIds: GetPermissionIds = ({ client, resource, permissionLevel, access }) =>
+  client
+    .select('permission_id')
+    .table('permission')
+    .where({
+      ['permission_entity']: resource,
+      ['permission_level']: permissionLevel,
+      ['access_type']: access,
+    });
+
+const getRoleId: GetRoleId = async({ client, role }) =>
+  client
+    .select('access_role_id')
+    .table('access_role')
+    .where({ ['access_role_name']: role })
+    .then((res) => res[0].access_role_id);
+
+const permissionsInitialiser: PermissionsInitialiser = (config) => {
+  const client = knex(config);
+  return {
+    grantExisting: async ({ resource, permissionLevel, access, role }) => {
```

Comment:
This is just so I can confirm my understanding of what this does: Grants requested access level on resource to a role, if that combination of (resource x access x permission level) already exists.

Reply:
Yes